object_manager.go

```diff
@@ -441,6 +448,124 @@ func (objMgr *ObjectManager) CreateEADefinition(eadef EADefinition) (*EADefiniti
+
+	if name != "" {
+		ipv6network.Ea["Network Name"] = name
+	}
+	ref, err := objMgr.connector.CreateObject(ipv6network)
+	if err != nil {
+		return nil, err
+	}
+	ipv6network.Ref = ref
+
+	return ipv6network, err
+}
+
+func (objMgr *ObjectManager) GetIPv6Network(netview string, cidr string, ea EA) (*IPv6Network, error) {
+	if netview != "" && cidr != "" {
+		var res []IPv6Network
+
+		ipv6Network := NewIPv6Network(netview, cidr, ea)
+
+		if cidr != "" {
+			ipv6Network.Cidr = cidr
+		}
+
+		if ea != nil && len(ea) > 0 {
+			ipv6Network.eaSearch = EASearch(ea)
+		}
+
+		err := objMgr.connector.GetObject(ipv6Network, "", &res)
+
+		if err != nil || res == nil || len(res) == 0 {
+			return nil, err
+		}
+
+		return &res[0], nil
+	} else {
+		err := fmt.Errorf("Both network view and cidr values are required")
+		return nil, err
+	}
+}
+
+func (objMgr *ObjectManager) GetIPv6NetworkWithRef(ref string) (*IPv6Network, error) {
+	ipv6Network := NewIPv6Network("", "", nil)
+	err := objMgr.connector.GetObject(ipv6Network, ref, &ipv6Network)
+	return ipv6Network, err
+}
+
+func BuildIPv6NetworkFromRef(ref string) *IPv6Network {
+	// ipv6network/ZG5zLm5ldHdvcmskODkuMC4wLjAvMjQvMjU%3A2001%3Adb8%3Aabcd%3A0012%3A%3A0/64/global_view
+	r := regexp.MustCompile(`ipv6network/\w+:((([0-9a-fA-F]{1,4}%3A){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}%3A){1,7}%3A|([0-9a-fA-F]{1,4}%3A){1,6}%3A[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}%3A){1,5}(%3A[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}%3A){1,4}(%3A[0-9a-fA-F]{1,4}){1,3}([0-9a-fA-F]{1,4}%3A){1,3}(%3A[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}%3A){1,2}(%3A[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}%3A((%3A[0-9a-fA-F]{1,4}){1,6})|%3A((%3A[0-9a-fA-F]{1,4}){1,7}|%3A)|fe80%3A(%3A[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|%3A%3A(ffff(%3A0{1,4}){0,1}%3A){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}%3A){1,4}%3A((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))/\d+/(.+))`)
+
+	m := r.FindStringSubmatch(ref)
+
+	if m == nil {
+		return nil
+	}
+
+	return &IPv6Network{
+		Ref:         ref,
+		NetviewName: m[32],
+		Cidr:        m[1],
+	}
+}
+
+func (objMgr *ObjectManager) DeleteIPv6Network(ref string) (string, error) {
+	return objMgr.connector.DeleteObject(ref)
+}
+
+// updateEA updates the existing EA if a different value is passed
+// if the value does not exist, a new EA is created and added
+func (objMgr *ObjectManager) UpdateIPv6NetworkEA(ref string, updateEA EA, removeEA EA) (*IPv6Network, error) {
+	var res IPv6Network
+
+	nv := IPv6Network{}
+	nv.returnFields = []string{"extattrs"}
+	err := objMgr.connector.GetObject(&nv, ref, &res)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for k, v := range updateEA {
+		res.Ea[k] = v
+	}
+
+	for k := range removeEA {
+		_, ok := res.Ea[k]
+		if ok {
+			delete(res.Ea, k)
+		}
+	}
+
+	reference, err := objMgr.connector.UpdateObject(&res, ref)
+	res.Ref = reference
+	return &res, err
```

Comment:
Canonical way is to check if err != nil and return object only in case if err == nil.